### PR TITLE
Changes how telescience works and teleportation damage nerf

### PR DESCRIPTION
**Pull request by Alberyk**
This pr changes how telescience works; instead of just teleport things with zero risk or consequences, telescience will now create two portals, one in the telepad, and the other in the location.

Why?
Telescience has always been a really problematic part of the game, causing a lot of fuckery over the ages. It is pretty much riskless, besides some possible explosion, limb loss or whatever if you fuck up, but if you somewhat knew what you were doing, you could steal shit, and just teleport bombs or whatever upon your victims.

Now, telescience has some risk and difficulty attached to it. Since, for the scientist to steal/kidnapp/save someone, they must enter the portal, which puts them in the location of where they want to reach. It is also possible for the people that can reach the portal to end in the telescience lab, allowing for some counter play.

### CODE REVIEW

**Comment by ParadoxSpace** [comment]
Why not being able to place one portal at one location and one portal at another? It'd make it more useful.

**Comment by Arrow768** [comment]
> 
> 
> Why not being able to place one portal at one location and one portal at another? It'd make it more useful.

That would defeat the whole point of the PR

> It is also possible for the people that can reach the portal to end in the telescience lab, allowing for some counter play.

**Comment by VTCobaltblood** [comment]
Do the portals cause explosions when they're created in a solid tile?

**Comment by Alberyk** [comment]
Portals do not create explosions when they are created, but using them uses the regular teleport proc, which may cause explosions when something uses the portal.

**Comment by OneOneThreeEight** [comment]
>Now, telescience has some risk and difficulty attached to it.

Was the risk of losing a limb on impact **not** the killing blow to making telescience as inaccessible in usage by players as possible?

**Comment by LanceLynxx** [comment]
> pretty much riskless, besides some possible explosion, limb loss or whatever if you fuck up,

Yes I too find it riskless that experimenting with telesci can lead to losing limbs or dying.
I mean pfft, what's an arm or a leg in the name of science, right? _No risk._

If you're going this route, might as well remove telescience altogether. It's going to make it even more useless. Telepad will be just room decor.

Ever since the limb gib and object break on teleport PR, people barely touch it. Even I don't mess with it anymore.

I think a better idea is reverting to the original way it worked, and punish individuals who abuse it instead, since it's a player problem and not a mechanic problem/unbalance.

**Comment by Geevies** [comment]
I'd say, just replace the normal limb gibbing nerf with this. Imo, the gibbing was a drastic band-aid to a problem solved by this PR.

**Comment by Alberyk** [comment]
I changed so that teleporting into something solid deals 35 of brute, instead of instant gibbing the limb.

**Comment by skull132** [comment]
> If you're going this route, might as well remove telescience altogether. It's going to make it even more useless. Telepad will be just room decor.

Hyperboles like this are largely worthless. Don't bring them here.

> I think a better idea is reverting to the original way it worked, and punish individuals who abuse it instead, since it's a player problem and not a mechanic problem/unbalance.

No. It did not work with genetics, it will not work with Telescience.

**Comment by LanceLynxx** [comment]
> Hyperboles like this are largely worthless. Don't bring them here.
No, hyperboles are not worthless. Nerfing telescience (again) will dissuade people even more to play it, making it nothing more than a fluff piece of equipment, to the point of worthlessness.

> No. It did not work with genetics, it will not work with Telescience.
If you're taking genetics as the model, then follow through and delete telescience like it was done with genetics.

**Comment by skull132** [comment]
> If you're taking genetics as the model, then follow through and delete telescience like it was done with genetics.

Once again, appeals to hyper boles are worthless. And this here is why. Yes we _could_ just follow through with your hyperbole and delete telescience, as we did with genetics. Instead, we are attempting to represent it in a manner that is less of a thorn in our side from the perspective of abuse, while still **remaining useful**. If you legitimately think that developers have the time to spend tweaking mechanics so we could remove them later, _instead of actually removing them in the first place_, then there really isn't much discussion to be had further.

File: code/modules/telesci/telesci_computer.dm [hunk 1]
```diff
@@ -215,44 +210,22 @@
...
-					log_msg += ", "
-				var/x_offset = ROI.x - source.x
-				var/y_offset = ROI.y - source.y
-
-				do_teleport(ROI, locate(dest.x + x_offset, dest.y + y_offset, dest.z))
+
+			var/total_lifespawn = 250
+			var/total_failchance = 0
+
+			for(var/obj/item/I in crystals)
```
Comment (skull132) [inline code/modules/telesci/telesci_computer.dm]: Can't you just refactor this to
```
total_lifespawn += 25 * crystals.len
total_failchance += crystals.len
```

**Comment by MarinaGryphon** [comment]
Going back through a portal makes you phase into it and take damage.

**Pull request merged**